pkg/pillar/agentlog/agentlog.go

```diff
@@ -149,61 +151,131 @@ func (hook *SkipCallerHook) Levels() []logrus.Level {
-					}
-					sigUsr1File.Close()
-				} else {
-					log.Errorf("handleSignals: Error opening file %s with: %s", sigUsr1FileName, err)
-				}
-
-				usr1LogObject := base.EnsureLogObject(log, base.SigUSR1StacksType,
-					"", uuid.UUID{}, string(base.SigUSR1StacksType))
-				if usr1LogObject != nil {
-					log.Warnf("SIGUSR1 triggered with %d stacks", len(stackArray))
-					for _, stack := range stackArray {
-						usr1LogObject.Warnf("%v", stack)
-					}
-					log.Warnf("SIGUSR1: end of stacks")
-				}
+				dumpStacks(log, agentName)
 			case syscall.SIGUSR2:
-				log.Warnf("SIGUSR2 triggered memory info:\n")
-				sigUsr2File, err := os.OpenFile(sigUsr2FileName,
-					os.O_WRONLY|os.O_CREATE|os.O_SYNC|os.O_TRUNC, 0755)
-				if err != nil {
-					log.Errorf("handleSignals: Error opening file %s with: %s", sigUsr2FileName, err)
-				} else {
-					// This goes to /persist/agentdebug/<agentname>/sigusr2 file
-					sigUsr2File.WriteString("SIGUSR2 triggered memory info:\n")
-				}
-
-				logMemUsage(log, sigUsr2File)
-				logMemAllocationSites(log, sigUsr2File)
-				if sigUsr2File != nil {
-					sigUsr2File.Close()
-				}
+				go listenDebug(log, agentName)
 			}
 		}
 	}
 }
 
+func dumpMemoryInfo(log *base.LogObject, sigUsr2FileName string) {
+	log.Warnf("SIGUSR2 triggered memory info:\n")
+	sigUsr2File, err := os.OpenFile(sigUsr2FileName,
+		os.O_WRONLY|os.O_CREATE|os.O_SYNC|os.O_TRUNC, 0755)
+	if err != nil {
+		log.Errorf("handleSignals: Error opening file %s with: %s", sigUsr2FileName, err)
+	} else {
+		// This goes to /persist/agentdebug/<agentname>/sigusr2 file
+		sigUsr2File.WriteString("SIGUSR2 triggered memory info:\n")
+	}
+
+	logMemUsage(log, sigUsr2File)
+	logMemAllocationSites(log, sigUsr2File)
+	if sigUsr2File != nil {
+		sigUsr2File.Close()
+	}
+}
+
+func dumpStacks(log *base.LogObject, sigUsr1FileName string) {
+	stacks := getStacks(true)
+	stackArray := strings.Split(stacks, "\n\n")
+
+	sigUsr1File, err := os.OpenFile(sigUsr1FileName,
+		os.O_WRONLY|os.O_CREATE|os.O_SYNC|os.O_TRUNC, 0755)
+	if err == nil {
+		for _, stack := range stackArray {
+			// This goes to /persist/agentdebug/<agentname>/sigusr1 file
+			sigUsr1File.WriteString(stack + "\n\n")
+		}
+		sigUsr1File.Close()
+	} else {
+		log.Errorf("handleSignals: Error opening file %s with: %s", sigUsr1FileName, err)
+	}
+
+	usr1LogObject := base.EnsureLogObject(log, base.SigUSR1StacksType,
+		"", uuid.UUID{}, string(base.SigUSR1StacksType))
+	if usr1LogObject != nil {
+		log.Warnf("SIGUSR1 triggered with %d stacks", len(stackArray))
+		for _, stack := range stackArray {
+			usr1LogObject.Warnf("%v", stack)
+		}
+		log.Warnf("SIGUSR1: end of stacks")
+	}
+}
+
+func listenDebug(log *base.LogObject, agentName string) {
+	agentDebugDir := fmt.Sprintf("%s/%s/", types.PersistDebugDir, agentName)
+	stacksDumpFileName := agentDebugDir + "/sigusr1"
+	memDumpFileName := agentDebugDir + "/sigusr2"
+
+	mux := http.NewServeMux()
+
+	server := &http.Server{
+		Addr:    "localhost:6543",
```

Comment:
Previous discussion was here: https://github.com/lf-edge/eve/pull/3229#discussion_r1198880378

Now the code listens on localhost, but `eve http-debug` creates a new firewall rule to forward the port (@eriknordmark I am cheating here)

About adding it to edgeview: I don't know yet enough about it to have an opinion ... for now I'd like to add it to debugging and then perhaps later we can add it to edgeview.

Reply:
Can you not use it by ssh'ing into the device and accessing it using localhost?
If so I think we should put the remote access under edgeview since that enables policy controls and visbility in the controller.

Reply:
I added something to make it convenient with using via ssh: https://github.com/lf-edge/eve/pull/3237/files#diff-a3b42769797ef2e752af884f98fd0877b6034d918bae243c3f165d7d710987a1R169

Still looking into edgeview ...

Reply:
https://github.com/lf-edge/eve/pull/3237/commits/b92677cf5db3782cccf61dd2c99452adf4769a49 add the possibility to use pprof via edgeview